src/js/view/controls/rightclick.js

```diff
@@ -16,8 +16,9 @@ function createDomElement(html) {
 }
 
 export default class RightClick {
-    constructor(infoOverlay) {
```

Comment:
I don't see any usages for `this.controlbar`. Did you forget or is it not needed anymore?

Reply:
@rob Addressed your changes from the ticket in commercial . Everything is now in open source and instead of passing in the controlbar, i grab the ```sharing``` pliugin instance from the api. ```api.getPlugin('sharing')``` isnt available when we call ```new RightClick()``` so i pass the whole api object in. @johnBartos to your point earlier we could use a function here instead of passing the api here.

```javascript
new RightClick(this.infoOverlay, () => {
  api.getPlugin('sharing').open();
});
```

Reply:
I like the idea of passing in a `openSharing` callback handler. Ideally each item in right click could be organized with label, callback etc... and added in `rightclick.setup` not the constructor.